src/hotspot/share/opto/compile.cpp

```diff
@@ -2474,6 +2476,14 @@ void Compile::Optimize() {
     if (failing()) return;
   }
 
+  if (true) {
+    assert(!C->merge_stores_phase(), "merge store phase not yet set");
+    C->gather_nodes_for_merge_stores(igvn);
+    C->set_merge_stores_phase(true);
+    igvn.optimize();
+    C->set_merge_stores_phase(false);
+  }
+
```

Comment:
@vnkozlov @rwestrel 
1. Should I guard the optimization by a flag, maybe `MergeStores`?
2. Should I make a fresh pass over the whole graph like in `gather_nodes_for_merge_stores`, or rather have a list that collects the store nodes during igvn, and that I can just readily pick up here. Just like these lists:
https://github.com/openjdk/jdk/pull/16966/files#diff-f076857d7da81f56709da3de1511b1105727032186cde4d02c678667761f46eaR445-R451